packages/react-sdk/src/Provider.tsx

```diff
@@ -36,32 +43,83 @@ export class OptimizelyProvider extends React.Component<
   OptimizelyProviderProps,
   OptimizelyProviderState
 > {
-  sdkWrapper: UserWrappedOptimizelySDK
-
   constructor(props: OptimizelyProviderProps) {
     super(props)
+    const { optimizely, userId, userAttributes, user} = props
+
+    // check if user id/attributes are provided as props and set them ReactSDKClient
+    let finalUser: {
+      id: string
+      attributes: optimizely.UserAttributes
+    } | null = null
+
+    if (user) {
+      if ('then' in user) {
+        user.then(user => {
+          optimizely.setUser(user)
+        })
+      } else {
+        finalUser = {
+          id: user.id,
+          attributes: user.attributes || {},
+        }
+      }
+    } else if (userId) {
+      finalUser = {
+        id: userId,
+        attributes: userAttributes || {},
+      }
+      // deprecation warning
+      console.warn(
+        'Passing userId and userAttributes as props is deprecated, please switch to using `user` prop',
+      )
+    }
 
-    const { optimizely, userId, userAttributes } = props
-    this.sdkWrapper = createUserWrapper({
-      instance: optimizely,
-      userId,
-      userAttributes,
-    })
+    if (finalUser) {
+      optimizely.setUser(finalUser)
+    }
+  }
+
+  componentDidUpdate(prevProps: OptimizelyProviderProps): void {
+    if (prevProps.isServerSide) {
+      // dont react to updates on server
+      return
+    }
+    const { optimizely  } = this.props
+    if (this.props.user && 'id' in this.props.user) {
+      if (!optimizely.user.id) {
+        // no user is set in optimizely, update
+        optimizely.setUser(this.props.user)
+      } else if (
+        // if the users aren't equal update
+        !areUsersEqual(
+          {
```

Comment:
nit: Can't we just pass in `optimizely.user` here?

Reply:
Tried this, and it doesn't type check. Apparently, in this form, the compiler can tell that `optimizely.user.id` is truthy within this branch, but not so if i change the entire object to `optimizely.user`.

Reply:
Oh weird. Can't say we did not try!